### ISSUE TITLE
Exception by ps_facetedsearch

### ISSUE THREAD

**Issue by MathiasReker** [PrestaShop/PrestaShop#14801]
**Describe the bug**
ps_facetedsearch though exception on fresh install (fresh like in clean database by pscleaner module) if category contain products with properties.

**To Reproduce**
Install PrestaShop 1.7.6.0
Install PS Cleaner -> Detele categories, products etc.
Create a new product with properties.
Go to the category page of that product.
See exception by ps_facetedsearch

**Screenshots**
![Screenshot_2019-07-22 Screenshot](https://user-images.githubusercontent.com/26626066/61626123-b3902000-ac7c-11e9-9a3f-b84a17e6fd26.png)

**Additional information**
PrestaShop version: 1.7.6.0
PHP version: 7.2.19

Ping @PierreRambaud

**Comment by MathiasReker** [comment]
Screenrecord: https://reker.dk/video.webm

**Comment by khouloudbelguith** [comment]
Hi @MathiasReker,

I have the same issue with PS1.7.6.0 & ps_facetedsearchv3.1.0.
![image](https://user-images.githubusercontent.com/16067358/61627566-03201d00-ac78-11e9-92b5-785689f32420.png)
With PS1.7.6.0 & ps_facetedsearch v2.2.0. => OK.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by MathiasReker** [comment]
I am not 100% sure if the is a bug in ps_facetedsearch or if pscleaner clean too much data?

**Comment by MathiasReker** [comment]
@khouloudbelguith can you confirm following: Resetting ps_facetedsearch module fix the issue?

**Comment by MathiasReker** [comment]
![test](https://user-images.githubusercontent.com/26626066/61632087-4edcc180-ac8c-11e9-84a8-d6eda2076502.png)
If not working to reset the module try to delete this item.

I am not 100% sure what is the solution, because I tried different things.

**Comment by MathiasReker** [comment]
FYI @PierreRambaud

**Comment by khouloudbelguith** [comment]
@MathiasReker, I tried with another fresh install PS1.7.6.0, but it is OK.
I followed the same steps as before.
I attached a screen record
https://drive.google.com/file/d/1bvxJWftt_0t_GQOHQwEmQCgsdAu7sTns/view
Thanks to check & feedback.

**Comment by MathiasReker** [comment]
@khouloudbelguith Weird that you can no more reproduce the error. I think you might not have the module ps_facetedsearch activated.

**Comment by MathiasReker** [comment]
Btw. I think that the module store some connections to the old data and that is why resetting the module works. So maybe ps_facetedsearch should use the hook in pscleaner to reset those connections.

**Comment by khouloudbelguith** [comment]
@MathiasReker, no, it is enabled.

Thanks!

**Comment by PierreRambaud** [comment]
Not sure the faceted search module can be hooked on all modules :disappointed: 
But this is why we add a Clear Cache button. But we can maybe improve the php code to catch some errors :+1: 
Is it only occurred with pscleaner?

**Comment by MathiasReker** [comment]
@PierreRambaud I did only have the issue after using pscleaner. The clear cache button on the module does not solve the issue. I have to delete this item or reset the module.

![image](https://user-images.githubusercontent.com/26626066/61704760-b9513880-ad44-11e9-8aa3-e0a8bd0690a3.png)

**Comment by khouloudbelguith** [comment]
@MathiasReker, yes, after importing new products, the issue occurs again.
I need to delete the template to resolve the issue.

Thanks!

**Comment by PierreRambaud** [comment]
What the pscleaner or import are really doing? :scream: Are they changing features ids?

**Comment by MathiasReker** [comment]
Yes, the pscleaner does: https://github.com/PrestaShop/pscleaner/blob/master/pscleaner.php

**Comment by khouloudbelguith** [comment]
@PierreRambaud, the database is clean, first product created => id  == 1

**Comment by MathiasReker** [comment]
> 
> 
> @PierreRambaud, the database is clean, first product created => id == 1

Actually I don't see the problem in that. Maybe there should just be a warning i pscleaner saying that some features might need a reset?

**Comment by PierreRambaud** [comment]
If this module is changing all ids, many other modules must have a similar problem :scream:

**Comment by MathiasReker** [comment]
@PierreRambaud not in practice, because you will run pscleaner to erase demo data and then start configuring modules.

**Comment by PierreRambaud** [comment]
I try with pscleaner, the problem come from this module which is removing everything with sql queries instead of normal process. So many modules can have issues with it, like missing key in database, files still on disk, etc.
